### PR TITLE
77 add retry and handle promise reject

### PR DESCRIPTION
**Pull request by AbstractFruitFactory**
https://radixdlt.atlassian.net/secure/RapidBoard.jspa?rapidView=77&projectKey=RPNV1&modal=detail&selectedIssue=RPNV1-77&assignee=5da984885845970c32f07db3

### CODE REVIEW

File: src/modules/common/RadixUtil.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import Long from 'long'
 import BN from 'bn.js'
 import crypto from 'crypto'
+import { logger } from './RadixLogger'
```
Comment (MuncleUscles) [inline src/modules/common/RadixUtil.ts]: I don't think this is used anywhere?

File: src/modules/universe/RadixUniverse.ts [hunk 2]
```diff
@@ -224,7 +224,12 @@ export default class RadixUniverse {
                     })
 
                     nodeConnection.on('closed', () => {
-                        resolve(this.getNodeConnection(shard))
+                        this.getNodeConnection(shard)
+                            .then(connection => {
+                                resolve(connection)
+                            }).catch(e => {
+                                reject(e)
+                            })
```
Comment (MuncleUscles) [inline src/modules/universe/RadixUniverse.ts]: I think the new code is equivalent to the old code for all intents and purposes?

File: src/modules/txbuilder/RadixTransactionBuilder.ts [hunk 3]
```diff
@@ -776,6 +776,8 @@ export default class RadixTransactionBuilder {
             .then(connection => {
                 RadixTransactionBuilder.signAndSubmitAtom(atom, connection, signer)
                     .subscribe(stateSubject)
+            }).catch(e => {
+                logger.error(e)
```
Comment (MuncleUscles) [inline src/modules/txbuilder/RadixTransactionBuilder.ts:780]: I think this needs to do more than just catch and log it - we can easily propagate the error to the user through the stateSubject

**Review by MuncleUscles — COMMENTED** [review]
Looks good overall, just small comments

File: src/modules/common/RadixUtil.ts (hunk 1 above)
Comment (AbstractFruitFactory) [reply src/modules/common/RadixUtil.ts]: I removed it

File: src/modules/universe/RadixUniverse.ts (hunk 2 above)
Comment (alexmedkex) [reply src/modules/universe/RadixUniverse.ts:232]: Is it? Before, it would resolve even if there was a rejection from the recursive call

**Review by MuncleUscles — APPROVED** [review]

**Pull request merged**